tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_operations.go

```diff
@@ -70,14 +36,8 @@ func codeForOperation(operationName string, operation models.OperationDetails, r
 		longRunning = true
 	}
 
-	// TODO: refactor this to use the ObjectDefinition in `./models`
-	requestObject := ObjectDefinition{}
-	if operation.RequestObject != nil {
-		requestObject.ReferenceName = operation.RequestObject.ReferenceName
-		requestObject.Type = ObjectDefinitionType(operation.RequestObject.Type)
-
-	} else if strings.EqualFold(operation.Method, "POST") || strings.EqualFold(operation.Method, "PUT") {
-		// Post and Put operations should have one but it's possible they don't
+	if operation.RequestObject == nil && (strings.EqualFold(operation.Method, "POST") || strings.EqualFold(operation.Method, "PUT")) {
+		// Post and Put operations should have a RequestObject one but it's possible they don't
 		// TODO what goes here?
 	}
```

Comment:
This logic can be removed - it originally made sense but unfortunately the API Definitions prove otherwise